File: The Logic/RulesViewController.m
//
//  RulesViewController.m
//  The Logic
//
//  Created by babi4_97 on 05.11.2017.
//  Copyright © 2017 Ivan Babkin. All rights reserved.
//

#import "RulesViewController.h"

@interface RulesViewController () {
    
    __weak IBOutlet UIScrollView *_scrollView;
}
@end

@implementation RulesViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = @"Rules";
    CGFloat contentHeight = 0.0f;
    
    for (UIView *subview in _scrollView.subviews) {
        if (contentHeight < CGRectGetMaxY(subview.frame)) {
            contentHeight = CGRectGetMaxY(subview.frame);
        }
    }
    
    _scrollView.contentSize = CGSizeMake(CGRectGetWidth(_scrollView.frame), contentHeight);
}

- (void)viewWillAppear:(BOOL)animated {
    [self.navigationController setNavigationBarHidden:NO animated:animated];
    [super viewWillAppear:animated];
}

- (void)viewWillDisappear:(BOOL)animated {
    [self.navigationController setNavigationBarHidden:YES animated:animated];
    [super viewWillDisappear:animated];
}

@end

File: swift/The logic/RulesViewController.swift
//
//  RulesViewController.swift
//  The logic
//
//  Created by Ivan Babkin on 13.06.2018.
//  Copyright © 2018 Ivan Babkin. All rights reserved.
//

import UIKit

class RulesViewController: UIViewController {
    
    @IBOutlet weak var scrollView: UIScrollView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "Rules"
        
        var contentHeight: CGFloat = 0.0
        
        scrollView.subviews.forEach() { subview in
            contentHeight = max(contentHeight, subview.frame.maxY)
        }
        
        scrollView.contentSize = CGSize(width: scrollView.frame.width, height: contentHeight)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        navigationController?.setNavigationBarHidden(false, animated: true)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        navigationController?.setNavigationBarHidden(true, animated: true)
    }
}
